### PR TITLE
PCHR-2683: Leave Widget bug fixes

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
This PR fixes bugs related to Leave Widget.

## Problem 1
After adding new leave absence type it's displayed in leave widget when staff doesn't have entitlements for it.
dr__bob_adams___hr17-4

## Solution
In `leave-widget-absence-types-available-balance.component.js`, removing `absenceType` if it does not have entitlement.
dr__bob_adams___hr17-4

## Problem 2
Default leave types have higher priority in Next Leave section, they are displayed as next ones even when there are earlier requests of not default types.
dr__bob_adams___hr17-4

## Solution 2
In `leave-widget-next-leave.component.js`, the parameters were wrong in the API call, 
1. The sort params need to be sent as 3rd param of `API.all` call, but it was sent as part of 1st parameter.
2. We need to sort the requests in Ascending order to get the latest request first. But it was sorted in Descending order before.
dr__bob_adams___hr17-4

## Comments
`loads the leave request day types` test case was failing, fixed it.

### CODE REVIEW

**Comment by igorpavlov** [comment]
@deb1990 Please improve the title to this PR to make it a bit more specific. I suggest something like "PCHR-2683: Leave Widget bug fixes"

**Comment by igorpavlov** [comment]
> loads the leave request day types test case was failing, fixed it.

How come? Did it fail after you applied your changes or from start?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-absence-types-available-balance.component.js [hunk 1]
```diff
@@ -103,14 +103,17 @@ define([
      * is used to display the current balance for approved and open requestes.
      */
     function mapAbsenceTypesWithTheirEntitlements () {
-      vm.absenceTypeEntitlements = vm.absenceTypes.map(function (absenceType) {
+      vm.absenceTypeEntitlements = [];
+      _.each(vm.absenceTypes, function (absenceType) {
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-absence-types-available-balance.component.js]: Please add an empty line after `vm.absenceTypeEntitlements = [];`.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-widget/leave-widget-next-leave.component.spec.js [hunk 2]
```diff
@@ -133,11 +135,6 @@ define([
           expect(ctrl.nextLeaveRequest).toEqual(expectedNextLeave);
         });
 
-        it('converts the request dates from strings to Date objects', function () {
-          expect(moment.isMoment(ctrl.nextLeaveRequest.from_date)).toBe(true);
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-widget/leave-widget-next-leave.component.spec.js]: This test was testing the type, why it was deleted?

**Comment by deb1990** [comment]
> How come? Did it fail after you applied your changes or from start?
Yes it was failing before.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-widget/leave-widget-next-leave.component.spec.js (hunk 2 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-widget/leave-widget-next-leave.component.spec.js]: I added these two lines to fix the failing test case
```
expectedNextLeave.from_date = moment(expectedNextLeave.from_date);
expectedNextLeave.to_date = moment(expectedNextLeave.to_date);
```

So https://github.com/civicrm/civihr/pull/2296/files#diff-58556ef72eedbf30b19477254d60fa52R135 line takes care of the deleted test case.

**Pull request merged**